File: backend/app/schemas/mtf.py
from pydantic import BaseModel

from app.schemas.signal import RiskLevel, SignalAction, TechnicalIndicators


class TimeframeSignal(BaseModel):
    timeframe: str          # "1d" | "1w" | "1m"
    label: str              # "Short-term" | "Mid-term" | "Long-term"
    signal: SignalAction
    confidence: float
    reasoning: str
    indicators: TechnicalIndicators | None = None


class MTFSignalResponse(BaseModel):
    symbol: str
    combined_signal: SignalAction
    combined_confidence: float
    combined_reasoning: str
    risk_level: RiskLevel
    agreement: bool         # True if all 3 timeframes agree
    timeframes: list[TimeframeSignal]
    source: str = "rules"

File: frontend/src/types/mtf.ts
import type { RiskLevel, SignalAction, TechnicalIndicators } from "./signal";

export interface TimeframeSignal {
  timeframe: string;
  label: string;
  signal: SignalAction;
  confidence: number;
  reasoning: string;
  indicators?: TechnicalIndicators | null;
}

export interface MTFSignalResponse {
  symbol: string;
  combined_signal: SignalAction;
  combined_confidence: number;
  combined_reasoning: string;
  risk_level: RiskLevel;
  agreement: boolean;
  timeframes: TimeframeSignal[];
  source: string;
}
